core/arch/arm/mm/tee_pager.c

```diff
@@ -189,10 +189,38 @@ static tee_mm_entry_t *pager_alias_area;
  */
 static uintptr_t pager_alias_next_free;
 
-static uint32_t pager_lock(void)
+#ifdef CFG_TEE_CORE_DEBUG
+#define pager_lock(ai) pager_lock_dldetect(__func__, __LINE__, ai)
+
+static uint32_t pager_lock_dldetect(const char *func, const int line,
+				    struct abort_info *ai)
+{
+	uint32_t exceptions = thread_mask_exceptions(THREAD_EXCP_ALL);
+	unsigned int retries = 0;
+	unsigned int reminder = 0;
+
+	while (!cpu_spin_trylock(&pager_spinlock)) {
```

Comment:
I agree, it is confusing. It stems from `strex` returning `1` in `<Rd>` on failure and `0` on success and that `__cpu_spin_trylock()` tries to do the bare minimum.

The code here is correct though.